libs/community/tests/unit_tests/vectorstores/test_aerospike.py

```diff
@@ -0,0 +1,375 @@
+                    "foo": 0,
+                },
+            ),
+            call(
+                namespace="test",
+                key="1",
+                set_name="otherset",
+                record_data={
+                    "_id": "1",
+                    "text": "bar",
+                    "vector": [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0],
+                    "bar": 1,
+                },
+            ),
+        ]
+    )
+    mock_client.wait_for_index_completion.assert_called_once_with(
+        namespace="test",
+        name="dummy_index",
+    )
+
+
+def test_delete_returns_false(mock_client: MagicMock) -> None:
+    from aerospike_vector_search import AVSServerError
+
+    mock_client.delete.side_effect = Mock(side_effect=AVSServerError(rpc_error=""))
+    aerospike = Aerospike(
+        client=mock_client,
+        embedding=FakeEmbeddings(),
+        text_key="text",
+        vector_key="vector",
+        namespace="test",
+        set_name="testset",
+        distance_strategy=DistanceStrategy.COSINE,
+    )
+
+    assert not aerospike.delete(["foo", "bar"], set_name="testset")
+    mock_client.delete.assert_called_once_with(
+        namespace="test", key="foo", set_name="testset"
+    )
+
+
+def test_similarity_search_by_vector_with_score_missing_index_name(
+    client: Any,
+) -> None:
+    aerospike = Aerospike(
+        client=client,
+        embedding=FakeEmbeddings(),
+        text_key="text",
+        vector_key="vector",
+        # index_name="dummy_index",
+        namespace="test",
+        set_name="testset",
+        distance_strategy=DistanceStrategy.COSINE,
+    )
+
+    with pytest.raises(ValueError, match="index_name must be provided"):
+        aerospike.similarity_search_by_vector_with_score([1.0, 2.0, 3.0])
+
+
+def test_similarity_search_by_vector_with_score_filters_missing_text_key(
+    mock_client: MagicMock,
+) -> None:
+    from aerospike_vector_search.types import Neighbor
+
+    text_key = "text"
+    mock_client.vector_search.return_value = [
+        Neighbor(key="key1", fields={text_key: 1}, distance=1.0),
+        Neighbor(key="key2", fields={}, distance=0.0),
+        Neighbor(key="key3", fields={text_key: 3}, distance=3.0),
+    ]
+    aerospike = Aerospike(
+        client=mock_client,
+        embedding=FakeEmbeddings(),
+        text_key=text_key,
+        vector_key="vector",
+        index_name="dummy_index",
+        namespace="test",
+        set_name="testset",
+        distance_strategy=DistanceStrategy.COSINE,
+    )
+
+    actual = aerospike.similarity_search_by_vector_with_score(
+        [1.0, 2.0, 3.0], k=10, metadata_keys=["foo"]
+    )
+
+    expected = [
+        (Document(page_content="1"), 1.0),
+        (Document(page_content="3"), 3.0),
+    ]
+    mock_client.vector_search.assert_called_once_with(
+        index_name="dummy_index",
+        namespace="test",
+        query=[1.0, 2.0, 3.0],
+        limit=10,
+        field_names=[text_key, "foo"],
+    )
```

Comment:
**Security Issue**: The 'vector_search' method is called with incorrect field names, expecting 'text_key' and 'foo' to be used as metadata keys, which might not correspond to the actual data structure expected by the 'vector_search' method. <br> **Fix**: Ensure that the metadata keys passed to 'vector_search' match the expected keys in the data structure. <br> **Code Suggestion**: 
 ```
 - field_names=[text_key, "foo"]
 + field_names=["correct_field_name1", "correct_field_name2"]
 ```